rskj-core/src/main/java/co/rsk/peg/whitelist/WhitelistSupportImpl.java

```diff
@@ -0,0 +1,209 @@
+    }
+
+    @Override
+    public LockWhitelistEntry getLockWhitelistEntryByAddress(String addressBase58) {
+        try {
+            Address address = Address.fromBase58(btcContext.getParams(), addressBase58);
+
+            return whitelistStorageProvider.getLockWhitelist().get(address);
+        } catch (AddressFormatException e) {
+            logger.warn(INVALID_ADDRESS_FORMAT_MESSAGE, e);
+            return null;
+        }
+    }
+
+    /**
+     * Adds the given address to the lock whitelist.
+     *
+     * @param addressBase58    the base58-encoded address to add to the whitelist
+     * @param maxTransferValue the max amount of satoshis enabled to transfer for this address
+     * @return 1 SUCCESS, -1 ADDRESS_ALREADY_WHITELISTED, -2 INVALID_ADDRESS_FORMAT,
+     * -10 GENERIC_ERROR, 0 UNKNOWN_ERROR.
+     */
+    @Override
+    public int addOneOffLockWhitelistAddress(Transaction tx, String addressBase58, BigInteger maxTransferValue) {
+        try {
+            Address address = Address.fromBase58(btcContext.getParams(), addressBase58);
+            Coin maxTransferValueCoin = Coin.valueOf(maxTransferValue.longValueExact());
+            LockWhitelistEntry whiteListEntry = new OneOffWhiteListEntry(address, maxTransferValueCoin);
+
+            return this.addLockWhitelistAddress(tx, whiteListEntry);
+        } catch (AddressFormatException e) {
+            logger.warn(INVALID_ADDRESS_FORMAT_MESSAGE, e);
+            return INVALID_ADDRESS_FORMAT.getCode();
+        }
+    }
+
+    @Override
+    public int addUnlimitedLockWhitelistAddress(Transaction tx, String addressBase58) {
+        try {
+            Address address = Address.fromBase58(btcContext.getParams(), addressBase58);
+            LockWhitelistEntry whiteListEntry = new UnlimitedWhiteListEntry(address);
+
+            return this.addLockWhitelistAddress(tx, whiteListEntry);
+        } catch (AddressFormatException e) {
+            logger.warn(INVALID_ADDRESS_FORMAT_MESSAGE, e);
+            return INVALID_ADDRESS_FORMAT.getCode();
+        }
+    }
+
+    /**
+     * Removes the given address from the lock whitelist.
+     *
+     * @param addressBase58 the base58-encoded address to remove from the whitelist
+     * @return 1 SUCCESS, -1 ADDRESS_NOT_EXIST, -2 INVALID_ADDRESS_FORMAT, -10 GENERIC_ERROR,
+     * 0 UNKNOWN_ERROR.
+     */
+    @Override
+    public int removeLockWhitelistAddress(Transaction tx, String addressBase58) {
+        if (isAuthorizedLockWhitelistChange(tx)) {
+            return GENERIC_ERROR.getCode();
+        }
+
+        LockWhitelist whitelist = whitelistStorageProvider.getLockWhitelist();
+        try {
+            Address address = Address.fromBase58(btcContext.getParams(), addressBase58);
+
+            if (!whitelist.remove(address)) {
+                return ADDRESS_NOT_EXIST.getCode();
+            }
+            return SUCCESS.getCode();
+        } catch (AddressFormatException e) {
```

Comment:
See sonar complaint, log the exception

Reply:
SonarLint didn't show me anything. Where could I see that?

Reply:
Done!

Reply:
Install Sonar extension in Intellij, it will help you catch a lot of things while working on the code. Also always check the `Problems` tab

Reply:
Yes, I have installed SonarLint, and the Problems tabs didn't show me that warning. That's weird! Thanks so much, @marcos-iov!